### ISSUE TITLE
Should editor allow repositioning of a container's children?

### ISSUE THREAD

**Issue by ghost** [godotengine/godot#7773]
**Operating system or device - Godot version:**

This affects all Godot versions with GraphNode

**Issue description:**

Moving the GraphNode in the scene editor has no effect on how it is displayed at runtime. Furthermore, the position reset upon reopening the scene in the editor. Programmatically, set_pos has no effect, only set_offset has some effect.

**Steps to reproduce:**

Create a GraphNode in the editor under GraphEdit, move it, and run the scene.

**Link to minimal example project:**
:

**Comment by reduz** [comment]
yes, use set_offset() instead of set_pos(). GraphEdit is a container, so
like all containers it has control over their children positioning.

**Comment by tagcup** [comment]
Ah, makes sense. Thanks for the clarification!

**Comment by tagcup** [comment]
Oh, but wait. Other containers still don't reset their position in the editor when you reload the scene.

**Comment by bojidar-bg** [comment]
@tagcup Containers recalculate positions of their __children__. GraphEdit thus recalculates the positions of Graph**Node**s

**Comment by tagcup** [comment]
What I'm trying to say is, in the editor, you can actually change the position (it could be a button placed inside a vboxcontainer for example), and obviously, the parent isn't recalculating the position of the child, because otherwise, you wouldn't be able to move it in the editor in the first place.

So then, let me rephrase my question: if that positioning you make in the editor will be discarded anyway, why and how does the editor allow you to move it's children?

**Comment by reduz** [comment]
because containers don't work all the time, but when objects are
removed/added/resized, or when the container itself is resized.
It's just not expecting you to move objects around because you are supposed
to know it's a container when you use it.

**Comment by bojidar-bg** [comment]
@reduz Still, it is highly unintuitive, maybe we should disable dragging for container children?

**Comment by kubecz3k** [comment]
Not sure who opened this issue originally, but maybe you guys @bojidar-bg  @tagcup  know what's the status here?

**Comment by volzhs** [comment]
I think it's still in the same status so far.
I guess something needs to be done for this,
like what @bojidar-bg said, disable dragging for it
or repositioning to where it should be after dragging.
maybe @groud has some idea?

**Comment by groud** [comment]
Closing as superseded by #20750
